### ISSUE TITLE
apt-get upgrade fails due to makedev errors

### ISSUE THREAD

**Issue by drdivano** [tianon/docker-brew-ubuntu-core#80]
apt-get upgrade on Ubuntu 14.04 fails with the following errors:

```
Setting up makedev (2.3.1-93ubuntu2~ubuntu14.04.1) ...
mknod: 'mem-': Operation not permitted
makedev mem c 1 1 root kmem 0640: failed
mknod: 'kmem-': Operation not permitted
makedev kmem c 1 2 root kmem 0640: failed
mknod: 'null-': Operation not permitted
makedev null c 1 3 root root 0666: failed
mknod: 'port-': Operation not permitted
.... and so on....
```

Software versions:

```
# uname -r
3.13.0-106-generic

# docker version
Client:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 02:30:42 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.5
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   7392c3b
 Built:        Fri Dec 16 02:30:42 2016
 OS/Arch:      linux/amd64

```

Possible workaround:
Add the following line to Dockerfile:
`ENV container=lxc
`

(This disables creation of device files, as per https://bugs.launchpad.net/ubuntu/+source/makedev/+bug/1675163)

**Comment by tianon** [comment]
See https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#apt-get, especially:

> You should avoid `RUN apt-get upgrade` or `dist-upgrade`, as many of the “essential” packages from the base images won’t upgrade inside an unprivileged container. If a package contained in the base image is out-of-date, you should contact its maintainers. If you know there’s a particular package, `foo`, that needs to be updated, use `apt-get install -y foo` to update automatically.

**Comment by Adel-Magebinary** [comment]
removing makedev by apt remove makedev fixes the issue in JOYENT LCX Ubuntu 16.04 images.
